pkg/signup/signup_service_test.go

```diff
@@ -99,6 +100,32 @@ func (s *TestSignupServiceSuite) TestUserSignupInvalidName() {
 	require.Error(s.T(), err)
 }
 
+func (s *TestSignupServiceSuite) TestUserSignupCreateFails() {
+	svc, fake := newSignupServiceWithFakeClient()
+	fake.MockCreate = func(*v1alpha1.UserSignup) (*v1alpha1.UserSignup, error) {
+		return nil, errors.New("an error occurred")
+	}
+
+	userID, err := uuid.NewV4()
+	require.NoError(s.T(), err)
+
+	_, err = svc.CreateUserSignup(context.Background(), "jack.smith@redhat.com", userID.String())
+	require.Error(s.T(), err)
+}
+
+func (s *TestSignupServiceSuite) TestUserSignupGetFails() {
+	svc, fake := newSignupServiceWithFakeClient()
+	fake.MockGet = func(string) (*v1alpha1.UserSignup, error) {
+		return nil, errors.New("an error occurred")
+	}
+
+	userID, err := uuid.NewV4()
+	require.NoError(s.T(), err)
+
+	_, err = svc.CreateUserSignup(context.Background(), "hank.smith@redhat.com", userID.String())
+	require.Error(s.T(), err)
```

Comment:
Let’s check the error message too so we know that we got the expected error and not something else.

Reply:
Compared the error returned from the service with the one returned by the mock get function.